cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/ZKConfigUtils.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.kafka.cruisecontrol.config;
+
+import com.linkedin.kafka.cruisecontrol.config.constants.ExecutorConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.zookeeper.client.ZKClientConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class ZKConfigUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZKConfigUtils.class);
+
+    private static final Map<String, String> ZK_PROPERTIES = new HashMap<>();
+    static {
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_CNXN_SOCKET_CONFIG, ZKClientConfig.ZOOKEEPER_CLIENT_CNXN_SOCKET);
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_KEYSTORE_LOCATION_CONFIG, "zookeeper.ssl.keyStore.location");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_KEYSTORE_PASSWORD_CONFIG, "zookeeper.ssl.keyStore.password");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_KEYSTORE_TYPE_CONFIG, "zookeeper.ssl.keyStore.type");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_TRUSTSTORE_LOCATION_CONFIG, "zookeeper.ssl.trustStore.location");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_TRUSTSTORE_PASSWORD_CONFIG, "zookeeper.ssl.trustStore.password");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_TRUSTSTORE_TYPE_CONFIG, "zookeeper.ssl.trustStore.type");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_PROTOCOL_CONFIG, "zookeeper.ssl.protocol");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_ENABLED_PROTOCOLS_CONFIG, "zookeeper.ssl.enabledProtocols");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_CIPHER_SUITES_CONFIG, "zookeeper.ssl.ciphersuites");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "zookeeper.ssl.hostnameVerification");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_CRL_ENABLE_CONFIG, "zookeeper.ssl.crl");
+        ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_OCSP_ENABLE_CONFIG, "zookeeper.ssl.ocsp");
+    }
```

Comment:
Rather than hardcoding config names (e.g. `"zookeeper.ssl.keyStore.location"` and so on), can we refer to relevant config in `KafkaConfig` (e.g. `KafkaConfig.ZkSslKeyStoreLocationProp()` and so on)?

Reply:
Here is an example change that refers to the relevant config name in `KafkaConfig` rather than hardcoding the config name (i.e. just as you did in `ExecutorConfig`):

* Current version:
```
...
ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_KEYSTORE_LOCATION_CONFIG, "zookeeper.ssl.keyStore.location");
...
```

* Proposed version:
```
...
ZK_PROPERTIES.put(ExecutorConfig.ZOOKEEPER_SSL_KEYSTORE_LOCATION_CONFIG, KafkaConfig.ZkSslKeyStoreLocationProp());
...
```

Is it possible to update each `put` entry this way?

Reply:
The names on the right are the configurations ZooKeeper uses. This variable `ZK_PROPERTIES` maps Cruise Control configurations (this is what users will enter and I decided to reuse the same names as Kafka) to ZooKeeper configuration that are set on the ZooKeeper client.

Unfortunately, the ZooKeeper names don't seem to be available anywhere as constants. ZooKeeper only has `ZKClientConfig.ZOOKEEPER_CLIENT_CNXN_SOCKET`. Kafka also builds this mapping but it's not exposed either (https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaConfig.scala#L323-L337).